third_party/move/move-compiler-v2/tests/checking-lang-v2.2/lambda/constraints.move

```diff
@@ -0,0 +1,5 @@
+module 0xc0ffee::m {
+    struct Func1<phantom T>(|T|T) has copy, drop;
```

Comment:
Looks like we currently (incorrectly?) admit the following code:

```
    struct Func4<phantom T>(|&T|&T) has copy, drop;
```

We should add this test case here?